frontend/packages/console-app/src/components/file-upload/file-upload-context.ts

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { AlertVariant } from '@patternfly/react-core';
+import { NamespaceContext } from '@console/app/src/components/detect-namespace/namespace';
+import { useToast } from '@console/shared/src/components/toast';
+import {
+  ResolvedFileUpload,
+  isFileUpload,
+} from '@console/dynamic-plugin-sdk/src/extensions/file-upload';
+import { useResolvedExtensions } from '@console/dynamic-plugin-sdk/src/api/useResolvedExtensions';
+import {
+  getRequiredFileUploadExtension,
+  getAllSupportedFileExtensions,
+  isFileSupported,
+} from './file-upload-utils';
+
+export type FileUploadContextType = {
+  extensions: string[];
+  fileUpload: File;
+  setFileUpload: (file: File) => void;
+};
+
+export const FileUploadContext = React.createContext<FileUploadContextType>({
+  fileUpload: undefined,
+  setFileUpload: () => {},
+  extensions: [],
+});
+
+export const useValuesFileUploadContext = (): FileUploadContextType => {
+  const { t } = useTranslation();
+  const [fileUploadExtensions, resolved] = useResolvedExtensions<ResolvedFileUpload>(isFileUpload);
+  const toastContext = useToast();
+  const { namespace } = React.useContext(NamespaceContext);
+  const [file, setFile] = React.useState<File>(undefined);
+  const [fileExtensions, setFileExtensions] = React.useState<string[]>([]);
+
+  const setFileUpload = React.useCallback(
+    (f: File): void => {
+      if (fileExtensions.length > 0 && f) {
+        const isSupported = isFileSupported(f.name, fileExtensions);
+        if (isSupported) {
+          const requiredFileExtension = getRequiredFileUploadExtension(
+            fileUploadExtensions,
+            f.name,
+          );
```

Comment:
You only need to search for the matching extension. If it isn't found, then you have your answer for `isSupported`. No need to do an extension check followed by another lookup.